File: tasks/translation/output/Qwen/Qwen2.5-32B-Instruct-AWQ/codenet/cot/Java/Python/s010698290_comments_removed.py
class FastReader:
    def __init__(self):
        self.st = []
        self.br = iter(sys.stdin.read().split())

    def next(self):
        if not self.st:
            self.st = next(self.br).split()
        return self.st.pop(0)

    def nextInt(self):
        return int(self.next())

    def nextLong(self):
        return int(self.next())

    def nextDouble(self):
        return float(self.next())

    def nextLine(self):
        return next(self.br)


import sys

if __name__ == "__main__":
    try:
        s = FastReader()
        n = s.nextInt()
        m = s.nextInt()

        penal = [0] * (n + 1)
        solved = [False] * (n + 1)

        for i in range(m):
            num = s.nextInt()
            str = s.next()
            if str[0] == 'A':
                solved[num] = True
            elif not solved[num]:
                penal[num] += 1

        correct_ans = 0
        num_penal = 0
        for i in range(1, n + 1):
            if solved[i]:
                correct_ans += 1
                num_penal += penal[i]

        print(f"{correct_ans} {num_penal}")
    except Exception as e:
        print(e)

# End of Code

File: tasks/translation/output/Qwen/Qwen2.5-32B-Instruct-AWQ/codenet/cot/Java/C++/s010698290_comments_removed.cpp
#include <iostream>
#include <vector>
#include <string>
#include <sstream>

using namespace std;

class FastReader {
public:
    FastReader() {}

    string next() {
        while (st.empty()) {
            string line;
            getline(cin, line);
            stringstream ss(line);
            st = vector<string>();
            string token;
            while (ss >> token) {
                st.push_back(token);
            }
        }
        string token = st.front();
        st.erase(st.begin());
        return token;
    }

    int nextInt() {
        return stoi(next());
    }

    long long nextLong() {
        return stoll(next());
    }

    double nextDouble() {
        return stod(next());
    }

private:
    vector<string> st;
};

int main() {
    FastReader s;
    int n = s.nextInt();
    int m = s.nextInt();

    vector<int> penal(n + 1, 0);
    vector<bool> solved(n + 1, false);

    for (int i = 0; i < m; i++) {
        int num = s.nextInt();
        string str = s.next();
        if (str[0] == 'A') {
            solved[num] = true;
        } else if (!solved[num]) {
            penal[num]++;
        }
    }

    int correct_ans = 0;
    int num_penal = 0;
    for (int i = 1; i <= n; i++) {
        if (solved[i]) {
            correct_ans++;
            num_penal += penal[i];
        }
    }

    cout << correct_ans << " " << num_penal << "\n";

    return 0;
}

// End of Code
